src/deterministic.jl

```diff
@@ -205,9 +143,23 @@ function Deterministic(forecast::Deterministic, data::SortedDict{Dates.DateTime,
     return Deterministic(; vals...)
 end
 
+convert_data(data::AbstractDict{Dates.DateTime, T}) where {T <: Vector{<:Real}} =
+    SortedDict{Dates.DateTime, Vector{CONSTANT}}(data...)
+convert_data(data::AbstractDict{Dates.DateTime, T}) where {T <: Vector{<:Tuple}} =
+    SortedDict{Dates.DateTime, Vector{POLYNOMIAL}}(data...)
+convert_data(data::AbstractDict{Dates.DateTime, T}) where {T <: Vector{<:Vector{<:Tuple}}} =
+    SortedDict{Dates.DateTime, Vector{PWL}}(data...)
+
+# Workaround for a bug/limitation in SortedDict. If a user tries to construct
+# SortedDict(i => ones(2) for i in 1:2)
+# it won't discern the types and will return SortedDict{Any,Any,Base.Order.ForwardOrdering}
+# https://github.com/JuliaCollections/DataStructures.jl/issues/239
```

Comment:
@jd-lara FYI, you can read the comments in this issue.

Reply:
Will check